react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+
+class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      currentImageIndex: 0
+    };
+    this.handleClickRight = this.handleClickRight.bind(this);
+    this.handleClickLeft = this.handleClickLeft.bind(this);
+    this.handleClickDot = this.handleClickDot.bind(this);
+  }
+
+  mount() {
+    this.intervalId = setInterval(() => {
+      if (this.state.currentImageIndex === this.props.bbYoda.length - 1) {
+        this.setState({ currentImageIndex: -1 });
+      }
+      this.setState({ currentImageIndex: this.state.currentImageIndex + 1 });
+    }, 3000);
+  }
```

Comment:
This method should be named `componentDidMount`, you'll also want to make sure that the carousel loops as you expect without the user having to click anything